src/transport/include/antioch/mixture_averaged_transport_evaluator.h

```diff
@@ -579,12 +579,23 @@ namespace Antioch
         }
       case(MASS_FLUX_MASS_FRACTION):
         {
-          VectorStateType molar_fractions = zero_clone(mass_fractions);
-
-          mixture.X(mixture.M(mass_fractions),mass_fractions,molar_fractions);
-
+         
+VectorStateType molar_fractions = zero_clone(mass_fractions);
+	  mixture.X(mixture.M(mass_fractions),mass_fractions,molar_fractions);
+
+          //Clipping unrealistic values, and calculating MW_Mixture through the new mole fractions 
+	  typename value_type<VectorStateType>::type MW_Mixture = zero_clone(mass_fractions[0]);
+	  for(unsigned int s=0; s < D_vec.size(); s++)
+	    {
+	      if(molar_fractions[s] > 1)
+		molar_fractions[s] = 1;
+	      if(molar_fractions[s] < 1e-16)
+		molar_fractions[s] = 1e-16;
```

Comment:
Should we shout a warning here if we hit these cases?  This seems at first glance like the best thing to do, but imagine e.g. someone trying to take a derivative by central finite differencing around 0; adding clipping under the hood could turn the relative error from O(epsilon^2) to O(.5).

Reply:
Oh, I love the idea of a ```clip_values = false``` default argument.

Reply:
Yeah, most of this was just a quick fix to check that this was what was causing the discrepancies seen between antioch and cantera. The default argument would definitely be better.